include/deal.II/base/bounding_box.h

```diff
@@ -233,6 +255,13 @@ class BoundingBox
   void
   extend(const Number amount);
 
+  /**
+   * The same as above with the difference that a new BoundingBox instance is
+   * created without changing the current object.
+   */
+  BoundingBox<spacedim, Number>
+  extended(const Number amount) const;
```

Comment:
I would prefer `create_extended` or something like that; `extend` and `extended` are easy to confuse.